skademlia/table.go

```diff
@@ -110,11 +110,19 @@ func (t *Table) Update(target *ID) error {
 	if found := t.Find(b, target); found != nil {
 		b.Lock()
 
-		// address might differ for same public key (checksum
 		id := found.Value.(*ID)
-		id.address = target.address
+		// address might differ for same public key (checksum)
+		if id.address != target.address {
+			// address is different so we remove the existing ID, add the ID that has new address
```

Comment:
I think it might be helpful to expand this a bit to give slightly more context

```suggestion
			// address is different so we remove the existing ID, add the ID that has new address
			// we don't want to modify the object in place because we don't own it
			// so we replace it
```